### PR TITLE
Add module for IS as KeyManager

### PR DESCRIPTION
**Pull request by Savidude**

### CODE REVIEW

File: modules/apim_is_as_km/manifests/params.pp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  $local_member_host = $::ipaddress
+  $clustering_members = [
+    {
+      host_name => '127.0.0.1',
+      port => '4000'
+    }
+  ]
+
+  # Datasource configurations
+  $wso2identity_db_url = 'jdbc:h2:repository/database/WSO2AM_DB;DB_CLOSE_ON_EXIT=FALSE'
```
Comment (VimukthiPerera) [inline modules/apim_is_as_km/manifests/params.pp:67]: Why the Identity DB has the name WSO2AM_DB?

File: modules/apim_is_as_km/manifests/params.pp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  $wso2identity_db_username = 'wso2carbon'
+  $wso2identity_db_password = 'wso2carbon'
+  $wso2identity_db_driver = 'org.h2.Driver'
+  $wso2identity_db_validation_query = 'SELECT 1'
+
+  $wso2bps_db_url = 'jdbc:h2:file:repository/database/jpadb;DB_CLOSE_ON_EXIT=FALSE;MVCC=TRUE'
```
Comment (VimukthiPerera) [inline modules/apim_is_as_km/manifests/params.pp:73]: Is this the URL given as the default in the GA pack?

File: modules/apim_is_as_km/templates/carbon-home/repository/conf/registry.xml.erb [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    http://wso2.org/library/tutorials/2010/04/sharing-registry-space-across-multiple-product-instances
+    -->
+
+    <currentDBConfig>wso2registry</currentDBConfig>
+    <readOnly>false</readOnly>
+    <enableCache>true</enableCache>
+    <registryRoot>/</registryRoot>
+
+    <dbConfig name="govregistry">
+        <dataSource>jdbc/WSO2IdentityDB</dataSource>
```
Comment (VimukthiPerera) [inline modules/apim_is_as_km/templates/carbon-home/repository/conf/registry.xml.erb]: Shall we change this to Registry DB?

File: modules/apim_is_as_km/manifests/params.pp (hunk 1 above)
Comment (Savidude) [reply modules/apim_is_as_km/manifests/params.pp:67]: identity.xml points to the `jdbc/WSO2AM_DB` datasource by default. Therefore, we have set this as the URL to Identity DB by default

File: modules/apim_is_as_km/manifests/params.pp (hunk 2 above)
Comment (Savidude) [reply modules/apim_is_as_km/manifests/params.pp:73]: Yes

File: modules/apim_is_as_km/templates/carbon-home/repository/conf/registry.xml.erb (hunk 3 above)
Comment (Savidude) [reply modules/apim_is_as_km/templates/carbon-home/repository/conf/registry.xml.erb]: Done https://github.com/wso2/puppet-apim/pull/146/commits/31e10b2ea5e9aea78aa1f70d8c5117cfa010828f

**Review by VimukthiPerera — APPROVED** [review]

**Pull request merged**